hydromt/_utils/steps_validator.py

```diff
@@ -0,0 +1,25 @@
+import inspect
+from typing import TYPE_CHECKING, Any, Dict
+
+if TYPE_CHECKING:
+    from hydromt.models.model import Model
+
+from .rgetattr import rgetattr
+
+
+def validate_steps(model: "Model", steps: Dict[str, dict[str, Any]]) -> None:
+    for step, options in steps.items():
+        attr = rgetattr(model, step, None)
+        if attr is None:
+            raise AttributeError(f"Method {step} not found in model.")
+        if not getattr(attr, "__ishydromtstep__", False) == True:
+            raise AttributeError(
+                f"Method {step} is not allowed to be called on model, since it is not a HydroMT step definition."
+                " Add @hydromt_step if that is your intention."
+            )
+
+        # attribute found, validate keyword arguments
+        # Throws if bind fails.
+        sig = inspect.signature(attr)
+        options = options or {}
+        _ = sig.bind(**options)
```

Comment:
This code is sort of duplicate of what is happening in _validators/model_config.py. We should consider if we already want to merge this, or do that later when we have more clarity on the use of pydantic models in our code base.